src/lib.rs

```diff
@@ -4045,18 +3984,6 @@ impl Build {
             ))
         }
     }
-    fn is_wasi_target(target: &str) -> bool {
-        const TARGETS: [&str; 7] = [
-            "wasm32-wasi",
-            "wasm32-wasip1",
-            "wasm32-wasip1-threads",
-            "wasm32-wasip2",
-            "wasm32-wasi-threads",
-            "wasm32-unknown-wasi",
-            "wasm32-unknown-unknown",
-        ];
-        TARGETS.contains(&target)
-    }
```

Comment:
Cross-referencing https://github.com/rust-lang/cc-rs/issues/1126, I've replaced usage of `is_wasi_target` with `target.os == "wasi"` in this PR, but that of course won't catch `wasm32-unknown-unknown` - so there might now be places where instead of `target.os == "wasi"`, the correct check would be `target.arch == "wasm32" || target.arch == "wasm64"`

Reply:
Well...I think we need to do this, this change might break some use cases.